pkg/api/client.go

```diff
@@ -43,38 +33,37 @@ type Client interface {
 	Do(context.Context, *http.Request) (*http.Response, []byte, error)
 }
 
-// NewClient returns a new client for accessing API server; the supplied context is used for authentication/authorization
-// requests and the supplied transport (which may be nil in the case of the default transport) is used for all requests made
-// to the API server.
-func NewClient(ctx context.Context, cfg Config, transport http.RoundTripper) (Client, error) {
-	var err error
-
-	hc := &httpClient{}
-	hc.client.Timeout = 10 * time.Second
-
-	// Configure the OAuth2 transport
-	hc.client.Transport, err = cfg.Authorize(ctx, transport)
-	if err != nil {
-		return nil, err
-	}
-
-	// Configure the API endpoints
-	hc.endpoints, err = cfg.Endpoints()
+// NewClient returns a new client for accessing API server.
+func NewClient(address string, transport http.RoundTripper) (Client, error) {
+	u, err := url.Parse(address)
 	if err != nil {
 		return nil, err
 	}
-
-	return hc, nil
+	u.Path = strings.TrimRight(u.Path, "/")
+
+	return &httpClient{
+		endpoint: u,
+		client: http.Client{
+			Transport: transport,
+			Timeout:   10 * time.Second,
+		},
+	}, nil
 }
 
 type httpClient struct {
-	client    http.Client
-	endpoints func(string) *url.URL
+	endpoint *url.URL
+	client   http.Client
 }
 
 // URL resolves an endpoint to a fully qualified URL.
 func (c *httpClient) URL(ep string) *url.URL {
-	return c.endpoints(ep)
+	// IMPLICIT: c.endpoint.Path cannot end with "/" because it was stripped off in NewClient
+	// IMPORTANT: Do not use Path.Join as it will not properly preserve trailing slashes
+
```

Comment:
Resolution is different from joining paths: i.e. resolving `/bar` against `http://example.com/foo` is `http://example.com/bar` instead of `http://example.com/foo/bar` (which is what we are trying to get in this scenario). Originally this came about because we tried to match the OpenAPI document (i.e. the constant value is `/experiments/` instead of `experiments/` because OpenAPI "relative" paths must start with "/" which in that context makes more sense). I've bounced back and forth a few times on this and ended up here because I disliked it the least.